buildspec-linux-make-gcc-cov.yml

```diff
@@ -22,27 +22,56 @@ phases:
       - make -C unit CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
       - make -C jbmc/src CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
       - make -C jbmc/unit CXX="ccache g++" -j2 CPROVER_WITH_PROFILING=1
+      - echo Build completed on `date`
   post_build:
     commands:
-      - make -C unit test 
+      # Run unit test
+      - make -C unit test
+      - lcov --capture --directory . --output-file lcov.info
+      - lcov --remove lcov.info '/usr/*' '*/unit/*' --output-file lcov.info
+      # If $CODEBUILD_SOURCE_VERSION starts with 'pr/', filter out pr number, if not, return empty
+      - VCS_PULL_REQUEST=$(echo $CODEBUILD_SOURCE_VERSION | sed '/^pr\//!d;s/^pr\///')
+      # If $CODEBUILD_SOURCE_VERSION is commit id, set $VCS_COMMIT_ID
+      - VCS_COMMIT_ID=$(echo $CODEBUILD_SOURCE_VERSION | sed -r '/^[a-z0-9]{40}$/!d') 
+      - COV_SCRIPT=/root/.cache/codecov.sh
+      - if [ ! -f "$COV_SCRIPT" ]; then curl -s https://codecov.io/bash > "$COV_SCRIPT"; fi
+      # -f lcov.info is needed to avoid codecov.sh deleting gcno files includeing word 'coverage'. 
+      # -F is flag associate with the coverage report.
+      - bash $COV_SCRIPT -t "$CODECOV_TOKEN" -f lcov.info -c -F reg || true
+      # reset all execution counts to
+      - lcov --zerocounters --directory . 
+      # Run regression tests
       # Also enable compilation with coverage/profiling flags when running the
       # regression tests, to also get results for the driver program in
       # regression/invariants
       - make -C regression test CPROVER_WITH_PROFILING=1
+      - lcov --capture --directory . --output-file lcov.info
```

Comment:
:bulb: Since this is quite a lot of script in a travis file, it might be worth pulling out a `.sh`ell script that does this